src/main/java/be/seeseemelk/mockbukkit/entity/PlayerMockFactory.java

```diff
@@ -1,68 +1,37 @@
 package be.seeseemelk.mockbukkit.entity;
 
-import java.util.HashSet;
-import java.util.Random;
-import java.util.Set;
-import java.util.UUID;
-
 import be.seeseemelk.mockbukkit.ServerMock;
 
-public final class PlayerMockFactory
-{
-	private static final String[] FIRST_NAMES = { "James", "Mary", "John", "Particia", "Robert", "Jennifer", "Michael", "Elizabeth", "William", "Linda" };
-	private static final String[] LAST_NAMES = { "Smith", "Johnson", "Jones", "Brown", "Davis", "Miller", "Wilson", "Moore", "Taylor", "Anderson" };
+import java.util.Random;
+import java.util.UUID;
 
+public final class PlayerMockFactory {
 	private final ServerMock server;
-	private Random random = new Random();
-	private Set<String> usedNames = new HashSet<>();
+	private int currentNameIndex;
+	private final Random random = new Random();
 
-	public PlayerMockFactory(ServerMock server)
-	{
+	public PlayerMockFactory(ServerMock server) {
+		this.currentNameIndex = 0;
 		this.server = server;
 	}
 
-	/**
-	 * Generates a random name.
-	 *
-	 * @return A randomly generated name.
-	 */
-	private String getRandomName()
-	{
-		String firstName = FIRST_NAMES[random.nextInt(FIRST_NAMES.length)];
-		String lastName = LAST_NAMES[random.nextInt(LAST_NAMES.length)];
-		return firstName + " " + lastName;
-	}
-
 	/**
 	 * Gets a unique random name. Any name that is generated will never be generated again.
 	 *
 	 * @return A unique random name.
 	 */
-	private String getUniqueRandomName()
-	{
-		if (usedNames.size() >= 100)
-		{
-			throw new RuntimeException("Out of names");
-		}
-
-		while (true)
-		{
-			String name = getRandomName();
-			if (!usedNames.contains(name))
-			{
-				usedNames.add(name);
-				return name;
-			}
-		}
+	private String getUniqueRandomName() {
+		String name = "name_" + currentNameIndex++;
+		if (name.length() > 16) throw new RuntimeException("Maximum number of player names reached");
+		return name;
 	}
```

Comment:
We should also annotate this, also formatting.

I also feel like `name_` is not descriptive enough. I think "Player" + number is a bit more reasonable.
Then the use of `RuntimeException` here doesn't really seem verbose enough to me. Perhaps an `IllegalStateException`?
The docs also need to be updated too.

```suggestion
	private @NotNull String getUniqueRandomName()
	{
		String name = "Player" + currentNameIndex++;

		if (name.length() > 16)
		{
			throw new IllegalStateException("Maximum number of player names reached");
		}

		return name;
	}
```